fix: AI prompt breaks undo feature

syuzanvs opened the pull request:
- Preserved editor history by setting a marker when AI writing starts, allowing the undo command to revert to the state before AI content was generated.
- Updated undo and redo command handlers to check for AI writing state and prevent clearing history.
- Ensured that changes made during AI content insertion are marked as undoable, allowing users to revert to previous states.
- Improved the handling of AI-generated content to maintain a consistent user experience with the undo/redo functionality.

This commit addresses the issue where the undo button became unresponsive after generating AI content, ensuring that users can revert to the state before AI commands were executed.

Closes #85

jjroelofs commented on src/aiagentservice.ts:
```diff
@@ -504,7 +525,12 @@ export default class AiAgentService {
+		const viewFragment = editor.data.processor.toView( editorContent );
+		const modelFragment = editor.data.toModel( viewFragment );
+		editor.model.insertContent( modelFragment );
+		editor.editing.view.scrollToTheSelection();
```
The undo/redo seems to work great now! But why did you add a scrolling behavior? I think maybe this is some experimental code belonging to another PR?

It is causing unexpected scrolling behaviors after AI writing completes:

![Untitled Video January 13, 2025 6_15 PM](https://github.com/user-attachments/assets/e51c3fcb-1247-4e82-8197-5763c8775271)

syuzanvs replied:
@jjroelofs I was testing something and accidentally left the code there, we don't need it. It's removed

syuzanvs committed:
fix: AI prompt breaks undo feature

- Preserved editor history by setting a marker when AI writing starts, allowing the undo command to revert to the state before AI content was generated.
- Updated undo and redo command handlers to check for AI writing state and prevent clearing history.
- Ensured that changes made during AI content insertion are marked as undoable, allowing users to revert to previous states.
- Improved the handling of AI-generated content to maintain a consistent user experience with the undo/redo functionality.

This commit addresses the issue where the undo button became unresponsive after generating AI content, ensuring that users can revert to the state before AI commands were executed.

Closes #85

jjroelofs reviewed (approved):
great!
